plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java

```diff
@@ -560,65 +617,66 @@ public void checkAnnotations(final Statechart statechart) {
 
 	@Check(CheckType.NORMAL)
 	public void checkUnboundEntryPoints(final org.yakindu.sct.model.sgraph.State state) {
-		if (state.isComposite()) {
-			final List<Transition>[] transitions = STextValidationModelUtils
-					.getEntrySpecSortedTransitions(state.getIncomingTransitions());
-			Map<Region, List<Entry>> regions = null;
-
-			// first list contains Transitions without entry spec
-			if (!transitions[0].isEmpty()) {
-				regions = STextValidationModelUtils.getRegionsWithoutDefaultEntry(state.getRegions());
-				if (!regions.isEmpty()) {
-					for (Transition transition : transitions[0]) {
-						error(TRANSITION_UNBOUND_DEFAULT_ENTRY_POINT, transition, null, -1);
-					}
-					for (Region region : regions.keySet()) {
-						error(REGION_UNBOUND_DEFAULT_ENTRY_POINT, region, null, -1);
-					}
+		if (!state.isComposite()) {
```

Comment:
Here, the early return **would** be fine from an understandability point of view, **but** you changed the behaviour of the method - only the first block is only executed for composite States, the code below is executed for all states, or rather _was_.